client/my-sites/pages/page-list.jsx

```diff
@@ -229,8 +227,14 @@ var Pages = React.createClass( {
 			const site = this.props.sites.getSelectedSite();
 			const status = this.props.status || 'published';
 
-			if ( status === 'published' && get( site, 'options.show_on_front' ) === 'posts' ) {
-				rows.push( <BlogPostsPage key="blog-posts-page" /> );
+			if ( site && status === 'published' ) {
+				rows.push(
+					<BlogPostsPage
+						key="blog-posts-page"
+						homePageType={ site.options.show_on_front }
```

Comment:
I'm curious why the `lodash.get` guard against accessing undefined deep properties was done away with here (& not used for the `pageForPosts` prop below).

To clarify, I was not able to trigger an error by clearing localStorage or indexedDB.